ODIN_II/SRC/subtractions.cpp

```diff
@@ -655,6 +655,294 @@ void iterate_adders_for_sub(netlist_t* netlist) {
     return;
 }
 
+/**
+ *-------------------------------------------------------------------------
+ * (function: instantiate_single_bit_sub3)
+ *
+ * Clean up the memory by deleting the list structure of adders
+ *	during optimization
+ *-----------------------------------------------------------------------*/
+void instantiate_single_bit_sub3(nnode_t* node, short traverse_mark_number, netlist_t* netlist) {
+    /* validate input port sizes */
+    oassert(node->input_port_sizes[0] == 1);
```

Comment:
Can't quite figure out what this does from the comments and asserts.  Can you clarify?
Can anything a user types ever hit these asserts?  If so, should print error messages.

Reply:
You are right, this one was not commented on properly. I have changed the routine name and added more comments.